app/src/main/java/org/kiwix/kiwixmobile/zimManager/libraryView/adapter/LibraryViewHolder.kt

```diff
@@ -148,20 +148,17 @@ sealed class LibraryViewHolder<in T : LibraryListItem>(containerView: View) :
     ): Boolean =
       when (downloadState) {
         is DownloadState.Failed -> false
-        is DownloadState.Paused -> {
-          when (downloadState.reason) {
-            // Disable the pause button when the DownloadManager is waiting for
-            // Wi-Fi or network connection. This prevents the user from trying
-            // to resume the download, as it will not work without a connection.
-            Error.QUEUED_FOR_WIFI,
-            Error.WAITING_FOR_NETWORK -> false
-
-            else -> true
-          }
-        }
-
+        is DownloadState.Paused -> !shouldEnablePauseResumeButton(downloadState.reason)
         else -> true
       }
+
+    /**
+     * Disable the pause button when the DownloadManager is waiting for
+     * Wi-Fi or network connection. This prevents the user from trying
+     * to resume the download, as it will not work without a connection.
+     */
+    private fun shouldEnablePauseResumeButton(reason: Error?): Boolean =
```

Comment:
The method name should be `shouldEnablePauseResumeButtonForPauseReason` to clarify its functionality.

Reply:
To make the method more intuitive, we should check that the error does not exist in the list, rather than checking if it does. This adjustment ensures that the method's functionality directly reflects its name.


```kotlin
private fun shouldEnablePauseResumeButtonForPauseReason(reason: Error?): Boolean =
      reason !in listOf(Error.QUEUED_FOR_WIFI, Error.WAITING_FOR_NETWORK)
```

With this change, there's no need to negate the result in the calling code:

```kotlin
is DownloadState.Paused -> shouldEnablePauseResumeButtonForPauseReason(downloadState.reason)
```